File: FinanceApp.Api/Models/DTOs/AccountDTO.cs
using System.Collections.Generic;
using FinanceApp.Api.Models.Entities;

namespace FinanceApp.Api.Models.DTOs
{
    public class AccountDTO
    {
        public Account Account { get; set; }
        public IEnumerable<Bill> Bills { get; set; }
        public decimal BillSum { get; set; }
        public decimal? PayDeduction { get; set; }
        public decimal? PaycheckPercentage { get; set; }
        public decimal? ExpensesBeforeNextPaycheck { get; set; }
        public decimal? BalanceSurplus { get; set; }
        public decimal? RequiredSavings { get; set; }
    }
}

File: FinanceApp.Angular/src/app/accounts/shared/accountDTO.ts
import { Bill } from "src/app/bills/shared/bill";

export class AccountDTO {
    account: Account;
    bills: Bill[];
    billsSum: number;
    payDeduction: number;
    expensesBeforeNextPaycheck: number;
    paycheckPercentage: number;
    balanceSurplus: number;
    requiredSavings: number;
}

export class BillDTO {
    bill: Bill;
    payDeduction: number;
    paycheckPercentage: number;
    monthlyPercentage: number;
    requiredSavings: number;
}

export class DTO {
    accountDtos: AccountDTO[];
    billDtos: BillDTO[];
    sumOfAccountBalances: number;
    costOfBillsPerPayPeriod: number;
    monthlyCostOfBills: number;
    totalSurplus: number;
}